core/src/main/java/com/yahoo/oak/EntryArray.java

```diff
@@ -112,7 +112,7 @@ protected void clear() {
     // Entry marked deleted only off-heap is already considered deleted
     enum EntryState {
         /*
-         * The state of the entry is yet to be checked.
+         * The state of the entry is yet to be checked or Entry is empty
```

Comment:
I agree. There can be UNKNOWN for "not checked at all", and EMPTY for "checked, but nothing was there"

Reply:
Boris, can you please add the state EMPTY, with TODO comment. While the usage of this state will be differed to other PR. This way there are more chances this will not be forgotten.